tests/php/test-amp-twitter-embed.php

```diff
@@ -162,18 +152,19 @@ public function get_conversion_data() {
 	 *
 	 * @dataProvider get_conversion_data
 	 * @param string $source   Source.
-	 * @param string $expected_start Expected start.
-	 * @param string $expected_end   Expected end.
+	 * @param string $expected Expected content.
 	 */
-	public function test__conversion( $source, $expected_start, $expected_end = null ) {
+	public function test__conversion( $source, $expected ) {
 		$embed = new AMP_Twitter_Embed_Handler();
 		$embed->register_embed();
+
 		$filtered_content = apply_filters( 'the_content', $source );
+		$dom              = AMP_DOM_Utils::get_dom_from_content( $filtered_content );
+		$embed->sanitize_raw_embeds( $dom );
 
-		$this->assertStringStartsWith( $expected_start, $filtered_content );
-		if ( $expected_end ) {
-			$this->assertStringEndsWith( $expected_end, $filtered_content );
-		}
+		$content = AMP_DOM_Utils::get_content_from_dom( $dom );
+
+		$this->assertEquals( $content, $expected );
```

Comment:
@pierlon For PHPUnit assertion, the expected value always comes first, followed by the actual value.

This is important because PHPUnit will tell you what it expected, and what it got instead. If you put the values in the wrong order (like here), this is misleading and causes you to look in the wrong places.

I'm rectifying this in #3758, as I'm touching those tests anyway.

Reply:
Ah thanks for pointing that out, noted :+1: .